tutorials/screenrecorder.py

```diff
@@ -0,0 +1,129 @@
+"""
+    mslib.msui.screenrecorder
+    ~~~~~~~~~~~~~~~~~~~
+
+    This python script is meant for recording the screens while automated  tutorials are running.
+
+    This file is part of mss.
+
+    :copyright: Copyright 2021 Hrithik Kumar Verma
+    :copyright: Copyright 2021 by the mss team, see AUTHORS.
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+import numpy as np
+import cv2
+import datetime
+import os
+import sys
+import time
+import mss
+from PyQt5 import QtWidgets
+
+
+class ScreenRecorder:
+    """
+    This is the screen recording class which helps to record the screen.
+    """
+    def __init__(self):
+        """
+        The constructor sets the FPS, codecs, VideoWriter object, filename, and directory
+        for the recording of the screen.
+        """
+        self.record = True
+        self.app = QtWidgets.QApplication(sys.argv)
+        screen = self.app.primaryScreen()
+        self.width = screen.size().width()
+        self.height = screen.size().height()
+        self.fps = self.get_fps()
+        self.codec = cv2.VideoWriter_fourcc(*"mp4v")
+        current_time = datetime.datetime.now().strftime('%d-%m-%Y %H-%M-%S')
+        self.file_name = f'REC_{current_time}.mp4'
+        parent_dir = os.getcwd()
+        dir = "Screen Recordings"
+        try:
+            path = os.path.join(parent_dir, dir)
+            os.makedirs(path, exist_ok=True)
+            final_path = os.path.join(path, self.file_name)
+        except OSError:
+            print("Directory \"Recordings\" cannot be created")
+        self.recorded_video = cv2.VideoWriter(final_path, self.codec, self.fps, (self.width, self.height))
+        self.window_name = "Screen Recorder : MSS"
+        cv2.namedWindow(self.window_name, cv2.WINDOW_NORMAL)
+        cv2.resizeWindow(self.window_name, 480, 480)
+        cv2.moveWindow(self.window_name, self.width - 480, 0)
+
+    def get_fps(self):
```

Comment:
Your loop in here does not contain everything you have in your capture loop.
The FPS you get here are always higher than your computer is able to produce, since the capture loop also writes to the mp4 file and is thus slower.